osu.Game/Skinning/RulesetSkinProvidingContainer.cs

```diff
@@ -92,20 +87,12 @@ protected virtual void UpdateSkins()
                 }
             }
 
-            if (Ruleset.CreateResourceStore() is IResourceStore<byte[]> resources)
-            {
-                int defaultSkinIndex = SkinSources.IndexOf(skinManager.DefaultSkin);
+            int defaultSkinIndex = SkinSources.IndexOf(skinManager.DefaultSkin);
 
-                var rulesetResources = new ResourcesSkin(resources, host, audio);
-
-                if (defaultSkinIndex >= 0)
-                    SkinSources.Insert(defaultSkinIndex, rulesetResources);
-                else
-                {
-                    // Tests may potentially override the SkinManager with another source that doesn't include it in AllSources.
-                    SkinSources.Add(rulesetResources);
-                }
-            }
+            if (defaultSkinIndex >= 0)
```

Comment:
An explanation of why the ruleset's resources are to be added *before* the default skin (else last) is probably best added here. Something like:

```suggestion
            // Ruleset resources should be given the ability to override game-wide defaults
            // This is achieved by placing them before an instance of DefaultSkin.
            // Note that DefaultSkin may not be present in some test scenes.
            if (defaultSkinIndex >= 0)
```